include/libtorrent/alert_types.hpp

```diff
@@ -2663,11 +2663,27 @@ namespace libtorrent {
 		aux::allocation_slot m_v6_nodes_idx;
 	};
 
+	struct TORRENT_EXPORT block_uploaded_alert final : peer_alert
```

Comment:
Please document this alert (using comments on the lines above the declaration). It's worth mentioning that it's posted when a block is placed in the send buffer to a peer, which may or may not have been sent yet (and if the connection is closed, the alert may be posted without the bytes having been sent to the peer). It's also probably worth mentioning which category it belongs to.